### PR TITLE
Hotfix server crash

### PR DESCRIPTION
**Pull request by Raycoms**
Should fix the crash because of clientside only.

### CODE REVIEW

**Comment by Kostronor** [comment]
:+1: Can confirm crash is gone

**Comment by Raycoms** [comment]
:shipit:

File: src/main/java/com/minecolonies/util/BlockUtils.java [hunk 1]
```diff
@@ -106,4 +107,132 @@ public static boolean isWater(IBlockState iBlockState)
...
+            }
+            return Items.wheat_seeds;
+        }
+        else if(block instanceof BlockDaylightDetector)
+        {
+            return Item.getItemFromBlock(Blocks.daylight_detector);
+        }
+        else if(block instanceof BlockDoor)
+        {
+            return block == Blocks.iron_door ? Items.iron_door : (block == Blocks.spruce_door ? Items.spruce_door :
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/util/BlockUtils.java]: MAJOR Reduce the number of conditional operators (6) used in the expression (maximum allowed 4). [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1067)
MINOR ':' should be on a new line. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=checkstyle%3AOperator_Wrap)

File: src/main/java/com/minecolonies/util/BlockUtils.java [hunk 2]
```diff
@@ -106,4 +107,132 @@ public static boolean isWater(IBlockState iBlockState)
...
+                                                                  (block == Blocks.birch_door ? Items.birch_door :
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/util/BlockUtils.java]: MINOR ':' should be on a new line. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=checkstyle%3AOperator_Wrap)

File: src/main/java/com/minecolonies/util/BlockUtils.java [hunk 3]
```diff
@@ -106,4 +107,132 @@ public static boolean isWater(IBlockState iBlockState)
...
+                                                                   (block == Blocks.jungle_door ? Items.jungle_door :
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/util/BlockUtils.java]: MINOR ':' should be on a new line. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=checkstyle%3AOperator_Wrap)

File: src/main/java/com/minecolonies/util/BlockUtils.java [hunk 4]
```diff
@@ -106,4 +107,132 @@ public static boolean isWater(IBlockState iBlockState)
...
+                                                                    (block == Blocks.acacia_door ? Items.acacia_door :
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/util/BlockUtils.java]: MINOR ':' should be on a new line. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=checkstyle%3AOperator_Wrap)

File: src/main/java/com/minecolonies/util/BlockUtils.java [hunk 5]
```diff
@@ -106,4 +107,132 @@ public static boolean isWater(IBlockState iBlockState)
         return Objects.equals(iBlockState, Blocks.water.getDefaultState())
                || Objects.equals(iBlockState, Blocks.flowing_water.getDefaultState());
     }
+
+    /**
+     * Gets an item from a block.
+     * @param block input block.
+     * @return output item.
+     */
+    public static Item getItemFromBlock(Block block)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/util/BlockUtils.java]: MAJOR This method has 121 lines, which is greater than the 100 lines authorized. Split it into smaller methods. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS138)
MAJOR Reduce the number of returns of this method 29, down to the maximum allowed 5. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1142)
MAJOR The Cyclomatic Complexity of this method "getItemFromBlock" is 66 which is greater than 10 authorized. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

File: src/main/java/com/minecolonies/util/BlockUtils.java [hunk 6]
```diff
@@ -106,4 +107,132 @@ public static boolean isWater(IBlockState iBlockState)
...
+        {
+            return Items.sign;
+        }
+        else if(block instanceof BlockSkull)
+        {
+            return Items.skull;
+        }
+        else if(block instanceof BlockStem)
+        {
+            Item item =  block == Blocks.pumpkin ? Items.pumpkin_seeds : (block == Blocks.melon_block ? Items.melon_seeds : null);
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/util/BlockUtils.java]: MINOR Local variable 'item' could be declared final [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)

File: src/main/java/com/minecolonies/util/BlockUtils.java [hunk 7]
```diff
@@ -106,4 +107,132 @@ public static boolean isWater(IBlockState iBlockState)
...
+            return item != null ? item : null;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/util/BlockUtils.java]: MAJOR Avoid if (x != y) ..; else ..; [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3AConfusingTernary)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 11 issues
* MAJOR 6 major
* MINOR 5 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [BlockUtils.java#L19](https://github.com/Minecolonies/minecolonies/blob/d973bb7a1985fef2647f1371ed635ccf93522bcd/src/main/java/com/minecolonies/util/BlockUtils.java#L19): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 34 to the maximum authorized 30 or less. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1200)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 11 issues
* MAJOR 6 major
* MINOR 5 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [BlockUtils.java#L19](https://github.com/Minecolonies/minecolonies/blob/3310b5d0c3fea215642375143abc8b41b69937f3/src/main/java/com/minecolonies/util/BlockUtils.java#L19): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 34 to the maximum authorized 30 or less. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1200)

File: src/main/java/com/minecolonies/util/BlockUtils.java [hunk 8]
```diff
@@ -106,4 +107,140 @@ public static boolean isWater(IBlockState iBlockState)
...
+                item = Items.wheat_seeds;
+            }
+        }
+        else if(block instanceof BlockDaylightDetector)
+        {
+            item = Item.getItemFromBlock(Blocks.daylight_detector);
+        }
+        else if(block instanceof BlockDoor)
+        {
+            item = block == Blocks.iron_door ? Items.iron_door
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/util/BlockUtils.java]: MAJOR Reduce the number of conditional operators (6) used in the expression (maximum allowed 4). [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1067)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 4 issues
* MAJOR 4 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [BlockUtils.java#L19](https://github.com/Minecolonies/minecolonies/blob/cfb1dbc97009b378b99cef9831a6705d48c52fc8/src/main/java/com/minecolonies/util/BlockUtils.java#L19): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 34 to the maximum authorized 30 or less. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1200)

**Comment by Raycoms** [comment]
for me: :shipit:

**Comment by cltnschlosser** [comment]
:+1: for the crash fix, but this breaks mod support, we should find a Better way, and if there isn't one, we should not use instanceof if we are going to return exact instances, we should check for exact block instances.

**Pull request merged**